src/Core/Layer/GeometryLayer.js

```diff
@@ -0,0 +1,104 @@
+import { EventDispatcher } from 'three';
+import Picking from '../Picking';
+
+function GeometryLayer(id, object3d) {
+    if (!id) {
+        throw new Error('Missing id parameter (GeometryLayer must have a unique id defined)');
+    }
+    if (!object3d || !object3d.isObject3D) {
+        throw new Error('Missing/Invalid object3d parameter (must be a three.js Object3D instance)');
+    }
+
+    this._attachedLayers = [];
+    this.type = 'geometry';
+    this.protocol = 'tile';
```

Comment:
Is GeometryLayer restricted to be used with the TileProtocol?

Reply:
Actually, it shouldn't have a tile protocol in `geometryLayer`. you would need another `tiledGeometryLayer` or `tiledMeshLayer` inherited of `geometryLayer`.

Ideally it would be necessary to remove the layer protocol, and to replace it a source (where source is object which makes it possible to obtain the data)